dvc/objects/db/__init__.py

```diff
@@ -78,3 +79,12 @@ def __getattr__(self, name):
     def by_scheme(self):
         self._init_odb(self.CLOUD_SCHEMES)
         yield from self._odb.items()
+
+    @classmethod
+    def get_staging(cls):
```

Comment:
This is `get_staging()` method instead of a `staging` property since we will be changing the state of `odb.read_only` depending on where we are in the staging process (we need to be able to `odb.add()` while staging an object, then we want to mark the odb instance as read_only once staging is completed, so we don't try to push those objects later.

Note that fsspec MemoryFS is global, so while individual odb instances are marked read_only for internal DVC purposes, every instance always points to a single shared/global memfs storage